### ISSUE TITLE
Support an `elif` operator

### ISSUE THREAD

**Issue by ahal** [json-e/json-e#454]
I've seen many JSON-e templates get massive and extremely difficult to read (specifically in Taskcluster's `.taskcluster.yml` files). One reason for this is the lack of an `elif` or `elsif` clause which results in patterns like:
```
$if: val == "a"
then: 1
else:
    $if: val == "b"
    then: 2
    else:
         $if: val == "c"
         then: 3
... etc ...
```

Each arm adds another level of indentation and therefore makes the overall file more difficult to understand. I believe an `elif` or `elsif` clause would go a long way towards improving the readability of JSON-e templates.

**Comment by djmitche** [comment]
This is a cool idea.  Can you spell out what it might look like?  How does it compare to $match and $switch?

**Comment by ahal** [comment]
I think `$elif` would look like:
```
$if: val == "a"
then:
  1
$elif: val == "b"
then:
  2
else:
  3
```

I'll be totally honest that I didn't know about `$switch`! If we used that instead of multiple nested `$if/else` clauses I think it would solve most of problems I outlined in the description. However it looks like `$switch` doesn't support a default arm, so AFAICT the only way to implement a default currently is via `$if/else`. If we needed a default value, the `$switch` could be nested under an `$if/else` statement, though this would increase indentation again (but at least it would only be one level vs an arbitrary number of levels with pure `if/else`).
